yoyodyne/predict.py

```diff
@@ -104,23 +96,43 @@ def predict(
     util.log_info(f"Writing to {output}")
     _mkdir(output)
     loader = datamodule.predict_dataloader()
+    parser = datamodule.parser
+    mapper = data.Mapper(datamodule.index)
     with open(output, "w", encoding=defaults.ENCODING) as sink:
         if model.beam_width > 1:
             # Beam search.
             tsv_writer = csv.writer(sink, delimiter="\t")
             for predictions, scores in trainer.predict(model, loader):
                 predictions = util.pad_tensor_after_eos(predictions)
-                decoded_predictions = loader.dataset.decode_target(predictions)
-                row = itertools.chain(
-                    *zip(decoded_predictions, scores.tolist())
+                # TODO: beam search requires singleton batches and this
+                # assumes that. Revise if that restriction is ever lifted.
+                targets = [
+                    parser.target_string(mapper.decode_target(target))
```

Comment:
Just so I am clear, `mapper` goes from ints to strings, and `target_string` filters pads, etc and implements proper spacing? The method `target_string` is not very intuitive for me here.

Reply:
Yeah, it knows about the separators. That predates this PR.